directory.go

```diff
@@ -46,11 +40,8 @@ func (d directory) copyTo(dst string) error {
 		}
 	}
 
-	// Restore the directories modes if we made it writeable
-	if len(children) > 0 && d.info.Mode()&0200 == 0 {
-		if err := os.Chmod(dst,d.info.Mode());err != nil {
-			return err
-		}
+	if err := os.Chmod(dst,d.info.Mode());err != nil {
```

Comment:
Can you please `go fmt` the code? I believe `go fmt` will add spaces between the parameters and after the semicolon.

Reply:
Ah yes... thanks for the reminder... on most of my repos I've setup golangci-lint to aggressively police gofmt and other things using github actions.  If you are interested, I'd be happy to contribute a PR that does that here to (I am *super* into good static analysis including formating... but sadly, that's not how the code comes out of me without automated assistance :))

Reply:
Ha, absolutely. I also love golangci-lint and use it aggressively (now). I think at the time I wrote this I was using gometalinter, which has since been deprecated.

Reply:
Excellent!  I tend to tweek golangci-lint up to the max (enable basically everything at its strongest setting).

Next question... I've finding that on repeated use, copy errors rather than 'overlays' ... meaning if I use copy for:

if src is a/b/c - c being a file
copy.All(src,dst)
and then I add a/b/d
copy.All(src,dst) the second time errors out, because it hits c and the os.Create fails.  I've fixed that locally so that it simply overwrites c... is that behavior acceptable to you?

Reply:
Should be fixed now! :)